src/main/java/com/laytonsmith/core/functions/StringHandling.java

```diff
@@ -845,6 +846,184 @@ public Construct exec(Target t, Environment env, Construct... args) throws Cance
 	}
 
 	@api
+	public static class char_is_uppercase extends AbstractFunction {
+
+		@Override
+		public Class<? extends CREThrowable>[] thrown() {
+			return new Class[]{CREFormatException.class, CRECastException.class};
+		}
+
+		@Override
+		public String getName() {
+			return "char_is_uppercase";
+		}
+
+		@Override
+		public Construct exec(Target t, Environment environment, Construct... args) throws CancelCommandException, ConfigRuntimeException {
+			if(!(args[0].nval() instanceof String)) {
+			    throw new CREFormatException(this.getName() + " expects a string as first argument, but type " +
+				    args[0].typeof() + " was found.", t);
+			}
+			String text = args[0].nval();
+			
+			// Enforce the fact we are only taking the first character here
+			// Do not let the user pass an entire string. Only a single character.
+			if(text.length() > 1) {
+			    throw new CRECastException("Got " + text + " instead of expected character.", t);
+			}
+			
+			char check = text.charAt(0);
+			if(Character.isUpperCase(check)) {
+			    return CBoolean.get(true);
+			}
+			else {
+			    return CBoolean.get(false);
+			}
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{1};
+		}
+
+		@Override
+		public String docs() {
+			return "boolean {char} Determines if the character provided is uppercase or not.";
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return false;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return null;
+		}
+
+		@Override
+		public CHVersion since() {
+		    return CHVersion.V3_3_2;
+		}
+	}
+
+	@api
```

Comment:
These functions are incredibly specific. char_is_uppercase is fine, but what is the use case for adding this to the core? I would prefer to let users write a proc to do this.